### ISSUE TITLE
Naming/FileName with ExpectMatchingDefinition and CheckDefinitionPathHierarchy does not enforce in custom folders

### ISSUE THREAD

**Issue by grosser** [rubocop/rubocop#10220]
## Expected behavior

```yaml
Naming/FileName:
  ExpectMatchingDefinition: true
  CheckDefinitionPathHierarchy: true
```
and
```ruby
# something/bar.rb
module Foo
  class Bar
```
should be invalid because `something != Bar`

## Actual behavior

it is valid (nothing with `--debug` either)

## Steps to reproduce the problem

see expected

## RuboCop version

```
1.22.3 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 2.6.5 x86_64-darwin19)
```

## Misc
I tried poking around in `lib/rubocop/cop/naming/file_name.rb` but none of the methods in there are getting called if the filename matches the expected constant (also tried with --cache=false)

**Comment by grosser** [comment]
opened https://github.com/rubocop/rubocop/pull/10224 to fix
